lms/static/scripts/frontend_apps/utils/test/jstor-test.js

```diff
@@ -1,40 +1,73 @@
-import { articleIdFromURL } from '../jstor';
+import { articleIdFromUserInput } from '../jstor';
 
 describe('utils/jstor', () => {
-  describe('articleIdFromURL', () => {
+  describe('articleIdFromUserInput', () => {
     [
-      // GOOD URLS
+      // Plain article ID
+      ['1234', '1234'],
+      ['j.ctv125jfg.3.6', 'j.ctv125jfg.3.6'],
+
+      // DOI
+      ['10.2307/1234', '10.2307/1234'],
+      ['10.7591/j.ctt5hh13f.4', '10.7591/j.ctt5hh13f.4'],
+
       // URL with article ID
       ['https://www.jstor.org/stable/1234', '1234'],
+
       // Querystring ignored
       ['https://www.jstor.org/stable/1234?q=bananas&bar=baz', '1234'],
+
       // Trailing slash(es) ignored
       ['https://www.jstor.org/stable/1234/', '1234'],
       ['https://www.jstor.org/stable/1234//', '1234'],
+
       // http protocol OK
       ['http://www.jstor.org/stable/1234', '1234'],
+
       // DOI Prefx and DOI suffix
       ['https://www.jstor.org/stable/10.1086/508573', '10.1086/508573'],
       [
         'https://www.jstor.org/stable/10.1086.3333.9/508573',
         '10.1086.3333.9/508573',
       ],
 
-      // BAD URLS
+      // PDF URLs
+      ['https://www.jstor.org/stable/pdf/4101611.pdf', '4101611'],
+      [
+        'https://www.jstor.org/stable/pdf/10.7591/j.ctt5hh13f.4.pdf',
+        '10.7591/j.ctt5hh13f.4',
+      ],
+
+      // Proxied JSTOR URLs
+      ['https://www-jstor-org.someuni.edu/stable/1234', '1234'],
+      ['https://www.jstor.org.someproxy.usc.edu/stable/1234', '1234'],
+    ].forEach(([input, expected]) => {
+      it('returns article ID if input format is supported', () => {
+        assert.equal(articleIdFromUserInput(input), expected);
+      });
+    });
+
+    [
       // Not a URL
-      ['foo', null],
+      'foo-bar',
```

Comment:
The comment for this test "Not a URL" perhaps would be more accurately "Not a URL or ID"?